playbooks/graphql-schema-design.md

```diff
@@ -123,6 +135,24 @@ e.g. an `artworks` field _and_ an `artworks_connection`, simply choose one form
 - In rare cases an immediate list may be used, but this should only be done in cases where the list has few entries
   and doesn’t require pagination, i.e. fetch _all_ entries at once.
 
+## ID fields
+
+An ID field that refers to e.g. a database ID should be called something like `internalID`, it should _never_ be called
+just `id`, as that name is reserved for ‘Global Object Identification’ (further explained in the next section).
+
+Its type should be `ID!`, which is a custom string scalar meant to convey that the value is an identifier and is not
+nullable, as database IDs never are `null`.
+
+## Global Object Identification
+
+All GraphQL services should follow the [Global Object Identification specification](https://facebook.github.io/relay/graphql/objectidentification.htm). Due to schema stitching, for their IDs services should encode their own service ID for metaphysics to be able to resolve a node ID back to its upstream service.
+
+For instance, Exchange should encode an Order with ID 42 as follows:
+
+```
+Base64("Exchange:Order:42")
+```
+
```

Comment:
Maybe recommend always lowercasing everything for consistency? (though maybe non-number IDs might fail on that?)

Reply:
👍 for system names being lower cased for consistency. Type names should likely exactly reflect the type name in the schema and I don't think we should do anything to the `id`.